File: programming-concepts/iteration/visual-basic/while_name_check.vb
' Raspberry Pi Foundation
' Usage licensed under CC BY-SA 4
' Note: This file is designed to be copied out and compiled on your machine.
' To run this file you need to:
' 1. Copy the contents
' 2. Paste them into the Visual Basic IDE of your choice (Visual Studio, for example)
' 3. Compile the program
' 4. Run the program

Module Program

    ' The Main subroutine is the default entry point for a VB program
    Sub Main()
        Console.WriteLine("Please enter your name: ")
        Dim name As String = Console.ReadLine()

        While name <> "Bob"
            Console.WriteLine("Try again Bob")
            Console.WriteLine("Please enter your name: ")
            name = Console.ReadLine()
        End While

        Console.WriteLine("Hi Bob")
    End Sub

End Module

File: programming-concepts/iteration/python/while_name_check.py
# Raspberry Pi Foundation
# Developed as part of Ada Computer Science
# 
# Usage licensed under CC BY-NC-SA 4.0

def main():
    print("Please enter your name: ")
    name = input()

    while name != "Bob":
        print("Try again Bob")
        print("Please enter your name: ")
        name = input()

    print("Hi Bob")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()
